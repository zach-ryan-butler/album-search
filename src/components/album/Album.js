import React from 'react';
import PropTypes from 'prop-types';
import styles from './Album.css';
import { Link } from 'react-router-dom';

function Album({ obj }) {
  const coverArt = obj['cover-art-archive'].count ? `http://coverartarchive.org/release/${obj.id}/front` : 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSCe_V81Br1ClM5ezpV5aJkoq23Y1ecEcz-5b68AAfPE9Xw95w5Gg';
  return (
    <section className={styles.Album}>
      <Link to={`/songs/${obj.name}/${obj.title}/${obj.id}`}>{obj.title}</Link>
      <img src={coverArt} />
    </section>
  );
}

Album.propTypes = {
  obj: PropTypes.shape({
    name: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired,
    ['cover-art-archive']: PropTypes.shape({
      count: PropTypes.number.isRequired
    }).isRequired
  }).isRequired
};

export default Album;
