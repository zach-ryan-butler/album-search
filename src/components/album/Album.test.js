import React from 'react';
import { shallow } from 'enzyme';
import Album from './Album';

describe('Album component', () => {
  it('renders Album', () => {
    const wrapper = shallow(<Album 
      obj={{
        name:'my name',
        title: 'my title',
        id: 'some id',
        ['cover-art-archive']: {
          count: 12
        }
      }}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
