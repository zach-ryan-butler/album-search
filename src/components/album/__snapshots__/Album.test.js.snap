// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Album component renders Album 1`] = `
<section
  className="Album"
>
  <Link
    to="/songs/my name/my title/some id"
  >
    my title
  </Link>
  <img
    src="http://coverartarchive.org/release/some id/front"
  />
</section>
`;
